ttt-ttpe/src/main/java/com/skynav/ttpe/layout/LineLayout.java

```diff
@@ -293,13 +293,16 @@ else if (c == Characters.UC_OBJECT) {
             return new IgnoredControlRun(s);
         }
         boolean inBreakingWhitespace = Characters.isBreakingWhitespace(c);
+        WhitespaceState startingWhitespaceState = getWhitespaceState(s, iterator.getIndex());
         while ((c = iterator.next()) != CharacterIterator.DONE) {
             if (c == Characters.UC_OBJECT)
                 break;
             else if (isIgnoredControl(c))
                 break;
             else if (inBreakingWhitespace ^ Characters.isBreakingWhitespace(c))
                 break;
+            else if (inBreakingWhitespace && startingWhitespaceState == WhitespaceState.PRESERVE)
```

Comment:
On XML elements where `xml:space="preserve"`, if there are multiple subsequent newline characters in the text, that should be equivalent to multiple subsequent `<br>` elements. Otherwise, if `xml:space` is not `"preserve"`, it's okay to merge them.